### ISSUE TITLE
amqp ingestion plugin

### ISSUE THREAD

**Issue by marksteele** [OpenTSDB/opentsdb#218]
It would be great if we could get a plugin that connects to a queue on an AMQP broker and binds to an exchange+topic to receive JSON formatted inputs.

Something like:
{"metric": "cpu","timestamp":12355623,"value":123,"tags":["host":"1.2.3.4","plugin":"foo"]"}\n
{"metric": "mem","timestamp":12355623,"value":123,"tags":["host":"1.2.3.4","plugin":"foo"]"}\n
{"metric": "disk","timestamp":12355623,"value":123,"tags":["host":"1.2.3.4","plugin":"foo"]"}\n

(newlines added for clarity, would be nice to be able to push multiple metrics per AMQP message for efficiency)

I've built something similar to integrate with opentsdb (https://github.com/marksteele/collectd-amqp-graphite-opentsdb-gateway), however as there's already publisher to amqp, an ingestor would be a nice touch.

Bonus points if there's an option for toggling snappy compression on the AMQP messages.

**Comment by IDerr** [comment]
Don't think that's the job of OpenTSDB, to ingest data like that, we should keep HTTP and TCP as input, and use scripts/software like Nifi or a script in python, to send data over protocols as I said before..

What do you think @manolama ??

**Comment by johann8384** [comment]
A feature request for a plug-in is reasonable probably, could go either way. We have a few built in plugins for other things, including the telnet and http protocols.

**Comment by IDerr** [comment]
Yeah sure, I didn't saw that there's already Real Time Publishing part on the resources, so yeah it is reasonable. 
Thanks
